snakemake/deployment/conda.py

```diff
@@ -874,9 +874,20 @@ def check(self):
         self.file.check()
 
     def get_conda_env(self, workflow, env_dir=None, container_img=None, cleanup=None):
+        def apply_token(filepath):
+            if "HOSTING_PROVIDER_TOKEN" in os.environ:
+                token=os.environ["HOSTING_PROVIDER_TOKEN"]
+                if "github" in filepath:
+                    return filepath.replace("raw",f":{token}@raw")
+                elif "gitlab" in filepath:
+                    return filepath + f"&private_token={token}"
+                else:
+                    raise WorkflowError(
+                        f"Access to private repositories is only available for github.com and gitlab.org"
+                    )
```

Comment:
This "code duplication" I don't really understand. If the conda env path is coming from a github or gitlab resource, it should already contain the token from the implementation in GithubFile and GitlabFile.

Reply:
That's a leftover from back when I was struggling to figure this out. Removed.